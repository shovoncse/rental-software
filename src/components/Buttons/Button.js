import React from 'react';
import styles from './Button.module.css';

const Button = ({ title, subTitle }) => {
  return <></>;
};

export default Button;
