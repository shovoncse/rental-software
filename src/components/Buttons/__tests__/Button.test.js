import React from 'react';
import ReactDOM from 'react-dom';
import Button from './../Button';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'

afterEach(cleanup);
it('should render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button></Button>, div);
});

it('should render button correctly', () => {
  const { getByTestId } = render(<Button text="Click me"></Button>);
  getByTestId('button');
  expect(getByTestId('button')).toHaveTextContent('Click me');
});

it("matches snapshot", () => {
  const tree = renderer.create(<Button text="Book"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
})