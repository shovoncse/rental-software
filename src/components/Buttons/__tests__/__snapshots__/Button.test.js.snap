// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="button undefined"
  data-testid="button"
  onClick={[Function]}
>
  <span>
    Book
  </span>
</div>
`;
